embeddings/task/lightning_task/lightning_task.py

```diff
@@ -40,6 +42,8 @@ def fit(
         self.trainer = pl.Trainer(default_root_dir=str(self.output_path), **self.task_train_kwargs)
         try:
             self.trainer.fit(self.model, data)
+            self.current_epoch = self.trainer.current_epoch
+            self.best_epoch = self.current_epoch - self.trainer.early_stopping_callback.patience
```

Comment:
I think that it should be isolated from training. I would rewrite it as a separate callback, like early stopping and do not rely with other callback.
```python
BestEpochCallback(monitor="val/loss", mode="min")
```

More information on callbacks: https://pytorch-lightning.readthedocs.io/en/stable/extensions/callbacks.html

Reply:
https://pytorch-lightning.readthedocs.io/en/stable/common/early_stopping.html